ContactDetailsApi/V1/Controllers/ContactDetailsController.cs

```diff
@@ -41,5 +44,26 @@ public async Task<IActionResult> GetContactDetailsByTargetId([FromQuery] Contact
 
             return Ok(new GetContactDetailsResponse(contacts));
         }
+
+        /// <summary>
+        /// Soft Deletes contact details for a particular id
+        /// </summary>
+        /// <response code="200">Successfully Operation</response>
+        /// <response code="404">Contact not found for ID requested</response>
+        /// <response code="500">Internal Server Error</response>
+        [ProducesResponseType(typeof(ContactDetailsResponseObject), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpDelete]
+        [LogCall(LogLevel.Information)]
+        public async Task<IActionResult> DeleteContactDetailsByTargetId([FromQuery] ContactQueryParameter queryParam)
```

Comment:
This name is wrong here (and all the way down the call stack).
We are deleting by Id only (we just need the target id to get the correct record because the data uses a composite primary key). Deleting by target id would mean deleting **all** the contacts for the target.